exasol/python_extension_common/deployment/language_container_deployer.py

```diff
@@ -248,6 +249,23 @@ def generate_activation_command(self, bucket_file_path: str,
             f"ALTER {alter_type.value} SET SCRIPT_LANGUAGES='{new_settings}';"
         return alter_command
 
+    def _wait_container_upload_completion(self, bucket_file_path: str):
+        """
+        The function waits till the container is fully uploaded and operational on all nodes.
+        It creates and then subsequently deletes a simple UDF that checks for the presence of
+        a certain file. This UDF is created in the current schema of the pyexasol connection
+        if one is open. Otherwise, a temporary schema will be created.
+        """
+        upload_path = self._upload_path(bucket_file_path)
+        schema = get_schema(self._pyexasol_conn)
```

Comment:
I propose to move this logic to `temp_schema` and add an optional boolean parameter `reuse`.
Then we don't need to duplicate the code here, can always use a context and do not need to call `get_schema()` another time. See below.